corehq/apps/hqcase/views.py

```diff
@@ -91,8 +91,12 @@ def delete_cases(self, request, domain):
 def case_api(request, domain, case_id=None):
     if request.method == 'GET' and case_id:
         return _handle_individual_get(request, case_id)
-    if request.method == 'GET' and not case_id and domain_needs_search_index(domain):
-        return _handle_list_view(request)
+    if request.method == 'GET' and not case_id:
+        if domain_needs_search_index(domain):
+            return _handle_list_view(request)
+        return JsonResponse({
+            'error': "You need the case search index to use this feature"
```

Comment:
what if you added the `CASE_API_V0_6` flag to this list instead?

https://github.com/dimagi/commcare-hq/blob/ae9f57d878a71def07af3435e27052c20a2b3d04/corehq/pillows/case_search.py#L61-L65

Reply:
Initially I left it out thinking that there might be projects that want to use the case update API and not the case list, and we needn't force the case search index on them.  That said, I think adding more projects to the case search is probably in our best interest and might not be a real blocker, so I like that proposal.

Wouldn't I also need some logic to do the initial build of the case search index for the domain?  Like this `save_fn`:
https://github.com/dimagi/commcare-hq/blob/703310fd517e34ffc74ee1b42389f6620f05b960/corehq/toggles.py#L819-L825

It looks like the other feature flags don't do that automatically, although `ECD_MIGRATED_DOMAINS` appears to be part of a mgmt command that turns it on separately.

Reply:
Yeah, I think you could piggy back on that save_fn to build the initial index.